### PR TITLE
Message Encryption functions added.

### PR DESCRIPTION
**Pull request by AlexanderSuprunenko**

### CODE REVIEW

File: include/lib/graft/cryptmsg.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+/*!
+ * \brief decryptMessage - (reverse of encryptForBs) decrypts data for one of the recipients using his b secret key.
+ *
+ * \param message - data that was created by encryptForBs.
+ * \param bkey - secret key corresponding to one of Bs that were used to encrypt.
+ * \param data - resulting decrypted data.
+ * \return true on success or false otherwise
+ */
+bool decryptMessage(const std::string message, const crypto::secret_key& bkey, std::string& data);
```
Comment (EDDragonWolf) [inline include/lib/graft/cryptmsg.h]: message - const &?
data - rename to `result`?

File: src/lib/graft/cryptmsg.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    message.resize(msgSize);
+    cryptoMessageHead& head = *reinterpret_cast<cryptoMessageHead*>(message.data());
+    head.plainSize = data.size();
+    crypto::secret_key r;
+    crypto::generate_keys(head.R, r);
+    head.count = BkeysCount;
+    //chacha encrypt with x
+    encryptChacha(data.data(), data.size(), X.x, message.data()+msgHeadSize);
+    //fill xEntry for each B
+    const crypto::public_key* pB = Bkeys;
```
Comment (EDDragonWolf) [inline src/lib/graft/cryptmsg.cpp]: Why not use std::vector?

File: include/lib/graft/cryptmsg.h [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#include "crypto/crypto.h"
+
+namespace graft::cryptmsg {
```
Comment (mbg033) [inline include/lib/graft/cryptmsg.h]: maybe just "graft::crypto" or "graft::crypto_tools", "cryptmsg" sounds senseless

File: src/lib/graft/cryptmsg.cpp [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+  crypto::chacha8(reinterpret_cast<const char*>(cipher) + sizeof(iv), cipher_size - prefix_size, key, iv, reinterpret_cast<char*>(plain));
+}
+
+constexpr uint16_t cStart = 0xA5A5;
+constexpr uint16_t cEnd = 0x5A5A;
+
+#pragma pack(push, 1)
+
+//sessionX in decrypted form contains session key x and constants to check that the decryption was correct
+struct sessionX
```
Comment (mbg033) [inline src/lib/graft/cryptmsg.cpp]: code style: type name starts with capital letter

File: include/lib/graft/cryptmsg.h (hunk 3 above)
Comment (bitkis) [reply include/lib/graft/cryptmsg.h]: "graft::crypto_tools" sounds better

**Comment by AlexanderSuprunenko** [comment]
moved to GraftNetwork

**Pull request closed without merging**